### DISCUSSION TITLE
Force the use of only bluetooth for data transfer?

### DISCUSSION THREAD

**Discussion by wisplite** [google/nearby#2872] [Q&A] [answered]
As the title states, I am interested in using the Nearby Connections API in an app I'm working on, but it is important that the app should not use WiFi for any communication. Is there a way to tell the API not to upgrade to a WiFi or WiFi-direct connection for sending data? All data transfer should be handled exclusively by Bluetooth.

**Comment by Xlythe** [comment, marked as answer]
[ConnectionOptions.Builder](https://developers.google.com/android/reference/com/google/android/gms/nearby/connection/ConnectionOptions.Builder)#setLowPower(true) and #setConnectionType(ConnectionType.NON_DISRUPTIVE) should turn off most WiFi upgrades.

But these params are explicitly about use cases and not mediums. We try to avoid exposing direct control over how the data is being sent -- do you have a reason why it must be Bluetooth?

**Reply by wisplite** [reply]
Congestion issues. In the environment the app is designed for, there are a ton of competing devices, and WiFi tends to be flaky. If we use WiFi, there's a good chance it will cause issues with communication in the app and other devices around us.

Bluetooth, on the other hand, tends to work fine because its low signal strength limits how many devices can be competing with each other, and the rapid frequency hopping is more suited for proper scaling.

I'll take a look at what you said to change. Even if we have to manually disable WiFi on the devices, that's fine, as long as we can prevent it from automatically enabling it and upgrading, and it seems like what you sent should do that.
